drafts/zip-str4d-shielded-coinbase.rst

```diff
@@ -0,0 +1,162 @@
+  for block height ``height`` to pay active funding streams with a standard P2SH script,
+  is amended to only apply to active funding streams that target a transparent address at
+  ``height``.
+
+- In the coinbase transaction for block height ``height``, for every active funding stream
+  that targets a Sapling address at ``height``, the transaction MUST include at least one
+  Sapling output that pays exactly the funding stream's value to the target Sapling
+  address.
+
+
+Rationale
+=========
+
+The ZIP does not require that all coinbase must be shielded immediately from activation of
+the network upgrade, so that miners and mining pools may gradually migrate from their
+existing transparent addresses to Sapling addresses. This also simplifies the consensus
+rules, because there are funding streams that target transparent addresses, and thus it
+remains necessary for the time being to support them. A future ZIP will require all
+coinbase to be shielded immediately.
+
+Enforcing coinbase maturity at the consensus level for Sapling outputs would incur
+significant complexity in the consensus rules, because it would require special-casing
+coinbase note commitments in the Sapling commitment tree. The standard recommendation when
+spending a note is to select an anchor 10 blocks back from the current chain tip, which
+acts as a de-facto 10-block maturity on all notes, coinbase included. This might be
+proposed as a consensus rule in future.
+
+There is another reason for shielded coinbase maturity being unnecessary: when a rollback
+occurs that would cause a shielded coinbase output to disappear, it will also invalidate
+every shielded transaction that uses an anchor descending from the tree that the shielded
+coinbase output had been appended to. That is, all economic activity would be rolled back
+in addition to the shielded coinbase output disappearing, so there is no reason to make
+shielded coinbase a special-case when the same behaviour occurs in regular shielded notes
+already. In the transparent coinbase case, only direct child transactions of the
+transparent coinbase would become invalid, and thus it would be possible to end up in a
+situation where a logical child transaction (for example, a mining pool paying out miners)
+persists in the block chain after its logical parent (the mining pool receiving a block)
+disappears.
+
+Requiring that note commitments are valid when recovering using a fixed outgoing viewing
+key implies that target addresses and values for all Sapling outputs within the coinbase
+are revealed. This is necessary to enforce funding streams correctly, and it is simpler to
+enforce this on all outputs. Additionally, this maintains the ability for network
+observers to track miners and mining pools. Meanwhile, the miners and mining pools could
+put useful or identifying text in the memo fields of the outputs, instead of storing it
+ad-hoc elsewhere in the coinbase transaction.
```

Comment:
Is this behavior obvious to everyone? I think it is. Would be a nasty surprise if not.